interchange/DeviceResources.capnp

```diff
@@ -203,9 +203,20 @@ struct Device {
   ######################################
   # Inter-site routing resources
   ######################################
+
+  enum WireCategory {
```

Comment:
I wonder if it is generally better to support a list of "arbitrary tags" on wires?

Reply:
I like that idea, although I don't know whether that should be in addition to or combined with this. Arbitrary tags on any objects could well be useful.

Reply:
As UltraScale clocking has a hierarchy of special interconnect types, would this be the right place to annotate that information here?  If we add a list of 'arbitrary tags', we vastly increase the support envelop, but that makes it harder for the tools to be data driven.  

Perhaps we augment the categories below so they can contain an optional tag with more specific info, for example:

```
    # general interconnect, usually with many uphill and downhill pips and spanning multiple tiles
    general : union {
        general @0 : Void;
        general_type @1: StringIdx $stringRef();
    }
    # pin/local wires, carry chains, dedicated paths, everything else
    special : union {
        special @2 : Void;
        special_type @3: StringIdx $stringRef();
    }
    # the global clock network
    global : union {
        global @2 : Void;
        global_type @3: StringIdx $stringRef();
    }
```
There is a similar construct used for [VCC/GND net names](https://github.com/SymbiFlow/fpga-interchange-schema/blob/main/interchange/DeviceResources.capnp#L402-L412).

Reply:
Yes, that seems like a good idea. I will do some more experimentation tomorrow with how this could work in practice but it looks in the right direction to me

Reply:
Does the idea of having an optional tag that may hold more info is still valid?

Reply:
This can be encoded in the wire type name/index, to some extent (eg crossreferencing), and we can now easily add extra fields to wire types as we come up with more requirements.